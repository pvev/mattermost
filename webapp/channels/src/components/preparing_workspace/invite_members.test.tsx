// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React, {ComponentProps} from 'react';
import {render, screen, fireEvent} from '@testing-library/react';
import {withIntl} from 'tests/helpers/intl-test-helper';

import InviteMembers from './invite_members';

describe('InviteMembers component', () => {
    let defaultProps: ComponentProps<any>;

    beforeEach(() => {
        defaultProps = {
            disableEdits: false,
            browserSiteUrl: 'https://my-org.mattermost.com',
            formUrl: 'https://my-org.mattermost.com/signup',
            teamInviteId: '1234',
            className: 'test-class',
            configSiteUrl: 'https://my-org.mattermost.com/config',
            onPageView: jest.fn(),
            previous: <div>{'Previous step'}</div>,
            next: jest.fn(),
            show: true,
            transitionDirection: 'forward',
        };
    });

    it('should match snapshot', () => {
        const component = withIntl(<InviteMembers {...defaultProps}/>);
        const {container} = render(component);
        expect(container).toMatchSnapshot();
    });

    it('renders invite URL', () => {
        const component = withIntl(<InviteMembers {...defaultProps}/>);
        render(component);
        const inviteLink = screen.getByTestId('shareLinkInput');
        expect(inviteLink).toHaveAttribute(
            'value',
            'https://my-org.mattermost.com/config/signup_user_complete/?id=1234',
        );
    });

    it('renders submit button with correct text', () => {
        const component = withIntl(<InviteMembers {...defaultProps}/>);
        render(component);
        const button = screen.getByRole('button', {name: 'Finish setup'});
        expect(button).toBeInTheDocument();
    });

    it('button is disabled when disableEdits is true', () => {
        const component = withIntl(
            <InviteMembers
                {...defaultProps}
                disableEdits={true}
            />,
        );
        render(component);
        const button = screen.getByRole('button', {name: 'Finish setup'});
        expect(button).toBeDisabled();
    });

    it('invokes next prop on button click', () => {
        const component = withIntl(<InviteMembers {...defaultProps}/>);
        render(component);
        const button = screen.getByRole('button', {name: 'Finish setup'});
        fireEvent.click(button);
        expect(defaultProps.next).toHaveBeenCalled();
    });
});
