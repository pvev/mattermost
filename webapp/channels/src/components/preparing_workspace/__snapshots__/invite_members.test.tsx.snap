// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteMembers component should match snapshot 1`] = `
<div>
  <div
    class="InviteMembers-body test-class"
  >
    <div
      class="SingleColumnLayout"
      style="width: 547px;"
    >
      <div>
        <div
          class="PageLine PageLine--no-left"
          style="margin-bottom: 50px; margin-left: 50px; height: calc(25vh);"
        />
        <div>
          Previous step
        </div>
        <h1
          class="PreparingWorkspaceTitle"
        >
          <span>
            Invite your team members
          </span>
        </h1>
        <p
          class="PreparingWorkspaceDescription"
        >
          <span>
            Collaboration is tough by yourself. Invite a few team members using the invitation link below.
          </span>
        </p>
        <div
          class="PreparingWorkspacePageBody"
        >
          <div
            class="InviteMembersLink"
          >
            <input
              aria-label="team invite link"
              class="InviteMembersLink__input"
              data-testid="shareLinkInput"
              readonly=""
              type="text"
              value="https://my-org.mattermost.com/config/signup_user_complete/?id=1234"
            />
            <button
              class="InviteMembersLink__button"
              data-testid="shareLinkInputButton"
            >
              <i
                class="icon icon-link-variant"
              />
              <span>
                Copy Link
              </span>
            </button>
          </div>
        </div>
        <div
          class="InviteMembers__submit"
        >
          <button
            class="primary-button"
          >
            <span>
              Finish setup
            </span>
          </button>
        </div>
        <div
          class="PageLine PageLine--no-left"
          style="margin-top: 50px; margin-left: 50px; height: calc(30vh);"
        />
      </div>
    </div>
  </div>
</div>
`;
